generators/app/index.js

```diff
@@ -549,23 +549,20 @@ module.exports = class extends BaseBlueprintGenerator {
 
             initGitRepo() {
                 if (!this.options['skip-git']) {
-                    this.isGitInstalled(code => {
-                        if (code === 0) {
-                            this.gitExec('rev-parse --is-inside-work-tree', { trace: false }, (err, gitDir) => {
-                                // gitDir has a line break to remove (at least on windows)
-                                if (gitDir && gitDir.trim() === 'true') {
-                                    this.gitInitialized = true;
-                                } else {
-                                    this.gitExec('init', { trace: false }, () => {
-                                        this.log(chalk.green.bold('Git repository initialized.'));
-                                        this.gitInitialized = true;
-                                    });
-                                }
-                            });
+                    if (this.gitInstalled || this.isGitInstalled()) {
+                        const gitDir = this.gitExec('rev-parse --is-inside-work-tree', { trace: false }).stdout;
+                        // gitDir has a line break to remove (at least on windows)
+                        if (gitDir && gitDir.trim() === 'true') {
+                            this.gitInitialized = true;
                         } else {
-                            this.warning('Git repository could not be initialized, as Git is not installed on your system');
+                            const shellStr = this.gitExec('init', { trace: false });
+                            this.gitInitialized = shellStr.code === 0;
+                            if (this.gitInitialized) this.log(chalk.green.bold('Git repository initialized.'));
+                            else this.warning(`Failed to initialize Git repository.\n ${shellStr.stderr}`);
                         }
-                    });
+                    } else {
+                        this.warning('Git repository could not be initialized, as Git is not installed on your system');
```

Comment:
Isn't git required? Should be error here.

Reply:
Fixed